keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,363 @@
+
+### Terminology
+
+* **PodNetwork** - name of the API object that will represent a network in
+  Kubernetes cluster
+* **Cluster Default PodNetwork** - This is the PodNetwork attached to the Pod
+  when no additional networking configuration is provided in Pod spec.
+* **Primary PodNetwork** - This is the PodNetwork inside the Pod which interface
+  is used for the default gateway.
+
+
+### User Stories
+
+All user stories represent the type of use cases the multi-networking API should
+be able to support. References to technologies or exact products does not
+indicate that this API will directly support them. User stories can be
+overlapping in some areas. We want to document the concrete use cases from an
+end-user standpoint -- they may (and should!) boil down to a few common
+primitives listed in the below requirements section.
+
+
+#### Story #1
+As a Cloud Native Network Function (CNF) vendor my workloads deals with 2 types
+of network traffic: control plane and dataplane. For regulatory compliance I have
+to isolate these 2 traffic types. To achieve this I require 2 separate network
+interfaces inside my Pod. The underlying implementation (outside the Pod) has to
+ensure isolation on either Layer-2 or Layer-3.
+
+<p align="center">
+  <img src="mn-story-1.png?raw=true" alt="multi-network story 1 network L2 isolation"/>
+</p>
+
+#### Story #2
+As a Cloud Native Network Function (CNF) vendor I require a HW-based interface
+(e.g. SRIOV VF) to be provisioned to my workload Pod. I need to leverage that HW
+for performance purposes (high bandwidth, low latency), that my user-space
+application (e.g. DPDK-based) can use. The VF will not use the standard netdev
+kernel module. The Pod’s scheduling to the nodes should be based on hardware
+availability (e.g. devicePlugin or some other way). This interface might not
+support some Kubernetes functionality like e.g. Services, I am willing to give
+that up for performance.
+
+<p align="center">
+  <img src="mn-story-2.png?raw=true" alt="multi-network story 2 HW interface"/>
+</p>
+
+#### Story #3
+I have implemented my Kubernetes cluster networking using a virtual switch. In
+this implementation I am capable of creating isolated Networks. I need a means
+to express to which Network my workloads connect to.
+
+<p align="center">
+  <img src="mn-story-3.png?raw=true" alt="multi-network story 3 virtual switch isolation"/>
+</p>
+
+#### Story #4
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I require multi-tenancy. The isolation has to be
+achieved on Layer-2 for security reasons.
+
+<p align="center">
+  <img src="mn-story-4.png?raw=true" alt="multi-network story 4 Kubevirt VMs"/>
+</p>
+
+#### Story #5
+As a platform operator I need to connect my on-premise networks to my workload
+Pods. I need to have the ability to represent these networks in my Kubernetes
+cluster in such a way that I can easily use them in my workloads.
```

Comment:
Representing a network is ok ... but how are pods going to be running on those networks if they are out of the cluster?

Reply:
That is up to the implementer.

Reply:
we need to define this, see gateway api per example that works solving this problem at the l4-l7  https://gateway-api.sigs.k8s.io/
![image](https://github.com/kubernetes/enhancements/assets/6450081/78ef8d9f-cc0e-45b1-abc2-05b814024280)

It defines the personas the use cases and the APIs and its behaviors to solve this problem, we need to do the same here , this is an important and complex topic we need to define